src/main/java/io/appform/dropwizard/actors/base/Handler.java

```diff
@@ -82,4 +101,43 @@ public void handleDelivery(String consumerTag, Envelope envelope,
     private MessageMetadata messageProperties(final Envelope envelope) {
         return new MessageMetadata(envelope.isRedeliver());
     }
+
+    public static Span buildChildSpan(AMQP.BasicProperties props, Tracer tracer) {
+        Tracer.SpanBuilder spanBuilder = tracer.buildSpan("receive")
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CONSUMER);
+
+
+        SpanContext parentContext = extract(props, tracer);
+        if (parentContext != null) {
+            spanBuilder.addReference(References.FOLLOWS_FROM, parentContext);
+        }
+
+        Span span = spanBuilder.start();
+        SpanDecorator.onResponse(span);
+
+        try {
+            if (props.getHeaders() != null) {
+                tracer.inject(span.context(), Format.Builtin.TEXT_MAP,
+                        new HeadersMapInjectAdapter(props.getHeaders()));
+            }
+        } catch (Exception e) {
+            // Ignore. Headers can be immutable. Waiting for a proper fix.
```

Comment:
didn;t get the comment

Reply:
this exception will arise incase someone tries to mutate the unmodifiable map -> props.getHeaders()

Reply:
now handled this properly